src/bin/html5-parser-test.rs

```diff
@@ -1,28 +1,40 @@
-use gosub_engine::testing::tree_construction::fixture_from_filename;
-use gosub_engine::testing::FIXTURE_ROOT;
+use gosub_engine::testing::tree_construction::fixture::{
+    get_fixture_root_path, read_fixture_from_path,
+};
+use gosub_engine::testing::tree_construction::Harness;
 use gosub_engine::types::Result;
 use std::io::Write;
-use std::path::Path;
+use std::path::{Path, PathBuf};
 use walkdir::WalkDir;
 
 fn main() -> Result<()> {
-    let mut files = get_files_from_path(format!("{}/tree-construction", FIXTURE_ROOT).as_str());
+    let mut files = get_files_from_path(get_fixture_root_path());
     files.sort();
 
     let mut total = 0;
     let mut failed = 0;
 
     for file in files.iter() {
-        let fixture = fixture_from_filename(file)?;
+        // if file != "math.dat" {
+        //     continue;
+        // }
+        let fixture = read_fixture_from_path(&get_fixture_root_path().join(file))?;
+
         print!("Test: ({:3}) {} [", fixture.tests.len(), file);
         let _ = std::io::stdout().flush();
 
+        let mut harness = Harness::new();
+
         // Run tests
-        for test in fixture.tests {
-            let results = test.run().expect("problem running tree construction test");
-            for result in results {
+        for test in fixture.tests.iter() {
+            for &scripting_enabled in test.script_modes() {
+                let result = harness
```

Comment:
We run a single test (loaded from the fixtures). It returns a result which holds all information which can be evaluated manually